subsystem/elevator.py

```diff
@@ -0,0 +1,204 @@
+import wpilib
+import rev
+from robotpy_toolkit_7407 import Subsystem
+from robotpy_toolkit_7407.motors.rev_motors import SparkMax, SparkMaxConfig
+from robotpy_toolkit_7407.utils.units import meters
+from robotpy_toolkit_7407.sensors.limit_switches.limit_switch import LimitSwitch
+import math
+import constants
+import config
+from wpimath.geometry import Pose3d
+# importing packages
+
+# TODO: check conversions
+
+SHOLDER_CONFIG = SparkMaxConfig(
+    0.2, 0, 0.003, 0.00015, (-0.5, 0.5), rev.CANSparkMax.IdleMode.kBrake
+)
+ELEVATOR_CONFIG = SparkMaxConfig(
+    0.00005, 0, 0.0004, 0.00017, idle_mode=rev.CANSparkMax.IdleMode.kBrake
+)
+
+
+class Elevator(Subsystem):  # elevator class
+    motor_extend: SparkMax = SparkMax(
+        config.elevator_motor_extend_id, config=ELEVATOR_CONFIG)  # motor that extends the arm
+    secondary_rotation_motor: SparkMax = SparkMax(
+        config.elevator_secondary_rotation_motor_id, inverted=True, config=SHOLDER_CONFIG)  # motor that rotates the arm
+    main_rotation_motor: SparkMax = SparkMax(
+        config.elevator_main_rotation_motor_id, inverted=False, config=SHOLDER_CONFIG)  # motor that rotates the arm
+    brake: wpilib.Solenoid = wpilib.Solenoid(
+        1, wpilib.PneumaticsModuleType.REVPH, config.elevator_brake_id)  # brake that holds the arm in place
+    initialized: bool = False
+    elevator_bottom_sensor = rev.CANSparkMax(config.elevator_main_rotation_motor_id, rev.CANSparkMax.MotorType.kBrushless).getReverseLimitSwitch(
+        rev.SparkMaxLimitSwitch.Type.kNormallyOpen)
+    elevator_top_sensor = rev.CANSparkMax(config.elevator_main_rotation_motor_id, rev.CANSparkMax.MotorType.kBrushless).getForwardLimitSwitch(
+        rev.SparkMaxLimitSwitch.Type.kNormallyOpen)
+    turn_sensor = rev.CANSparkMax(config.elevator_main_rotation_motor_id,
+                                  rev.MotorType.kBrushless).getAnalog()  # senses the rotation of the elevator
+    brake_enabled: bool = False
+    pose = Pose3d(0, 0, constants.pivot_point_height, 0, 0, 0)
+    elevator_length: float # the length of the elevator in meters
+    rotation_overide: bool = False  # if true, the rotation limits will be ignored
+    extension_overide: bool = False  # if true, the extension limits will be ignored
+    brake_overide: bool = False  # if true, the brake will not be enabled or disabled
+    intake_up: bool = True
+
+    def enable_brake(self):
+        if self.brake_overide:
+            return
+        else:
+            self.brake_enabled = True
+            self.brake.set(True)
+
+    def disable_brake(self):
+        if self.brake_overide:
+            return
+        else:
+            self.brake_enabled = False
+            self.brake.set(False)
+
+    def get_pose(self):
```

Comment:
Looking at `get_pose`, `set_pose` and `is_at_postion` what do you think will happen if we make these calls, in one loop before the elevator has had a chance to move? :-) 

```
elevator.set_pose(target)
elevator.is_at_position(target)
```